File: combine_basin_files.R
# restart computer
setwd("C:\\Users\\a_colleague\\Desktop\\basin_prcp_scripts\\data")

library(dplyr)
library(data.table)

pfiles <- list.files(pattern = 'basin')

pdata <- fread(pfiles[1])
colnames(pdata) <- c('day', 'prcp')
pdata <- round(pdata,2)
pdata$date <- as.numeric(pdata[,1][[1]])
pdata$date <- as.Date(pdata$date, origin = "1979-01-01")

for(i in 2:42){
  # i = 2
  year <- 1978 + i
  current <- fread(pfiles[i])

  colnames(current) <- c('day', 'prcp')
  current <- round(current,2)
  current$date <- as.numeric(current[,1][[1]])
  current$date <- as.Date(current$date, origin = paste(year, "-01-01", sep = ''))
  pdata <- rbind(pdata, current)
}

#plot annual total
pdata_annual <- group_by(pdata, year(date)) %>%
  summarize(total_prcp = sum(prcp))

plot(pdata_annual$`year(date)`, pdata_annual$total_prcp,
     xlab = 'Year', ylab = 'Annual precipitation, mm')

File: combine_basin_files.py
#restart computer 

import os 
import glob 
import pandas as pd 
from datetime import datetime

os.chdir("C:\\Users\\a_colleague\\Desktop\\basin_prcp_scripts\\data")

# get files from 1979-2021 

pfiles = glob.glob('basin*')

pdata = pd.read_csv(pfiles[1])
pdata['prcp'] = round(pdata['prcp'], 2)
pdata['day'] = pd.to_datetime(pdata['day'])

for i in range(1, 42):
    # i = 2
    year = 1978 + i 

    current = pd.read_csv(pfiles[i])
    current['prcp'] = round(current['prcp'], 2) 
    current['day'] = pd.to_datetime(current['day'])

    pdata = pdata.append(current)


pdata['year'] = pdata['day'].dt.year

# plot annual total 
pdata_annual = pdata.groupby(['year']).sum()

import matplotlib.pyplot as plt

plt.plot(pdata_annual.index, pdata_annual['prcp'], 'o', color = 'black')
plt.xlabel('Year')
plt.ylabel('Annual precipitation, mm')
